RFCs/FS-1092-Erased-Union-Types.md

```diff
@@ -0,0 +1,227 @@
+﻿# F# RFC FS-1092 - Erased Union Types
+
+This RFC covers the detailed proposal for this suggestion. [Erased type-tagged anonymous union types](https://github.com/fsharp/fslang-suggestions/issues/538).
+
+* [ ] Approved in principle
+* [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/538)
+* [ ] Details: TBD
+* [ ] Implementation: [Preliminary Prototype](https://github.com/dotnet/fsharp/pull/10566)
+
+
+# Summary
+[summary]: #summary
+
+Add erased union types as a feature to F#. Erased Union types provide some of the benefits of structural ("duck") typing, within the confines of a nominative type system.
+
+# Motivation
+[motivation]: #motivation
+
+Supporting erased union types in the language allows us to move more type information with the usual advantages this brings:
+
+* They serve as an alternative to function overloading.
+* They obey subtyping rules.
+* They allow representing subset of protocols as a type without needing to resort to the lowest common denominator like `obj`.
```

Comment:
DUs allow this today, too. What is the benefit of erased/ad-hoc unions as opposed to existing ones for this point? Perhaps this is worth expounding upon in an example later on.

Reply:
Am I allowed to add examples? I didn't want to step on anyone's toes.

Reply:
Either add a comment with your examples, or send a pull request to the pull request branch  :)